spec/facets/DatasetVersionDatasetFacet.json

```diff
@@ -0,0 +1,33 @@
+{
+  "$schema" : "https://json-schema.org/draft/2020-12/schema",
+  "$id" : "https://openlineage.io/spec/facets/1-0-0/DatasetVersionDatasetFacet.json",
+  "$defs" : {
+    "DatasetVersionDatasetFacet": {
+      "allOf": [
+        {
+          "$ref": "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
```

Comment:
do we want to add a field to capture what type of version it is? (iceberg, deltalake, ...)

Reply:
(possibly the wrong place to put this comment)

Reply:
Wouldn't we know that based on the dataset? If not, I'd prefer to have the dataset type at a higher level. Meaning, knowing the type may not just be useful for the _version_ facet, but potentially other facets we'll be defining in the future.

Reply:
We do have a `tableProvider` facet for Spark which contains information on that like iceberg on delta. I do not think we should duplicate this information. 

The only issue is that `tableProvider` is currently a custom facet within Spark integration only. Perhaps we should generalize it to something like `storageProvider` and promote to global spec. Anyway, this is out of the scope of this PR.

Reply:
That makes sense.
Yes, it sounds like a good idea to make it standard. Do you think this is already provided by DatasourceDatasetFacet?
https://github.com/OpenLineage/OpenLineage/blob/main/spec/facets/DatasourceDatasetFacet.json